### ISSUE TITLE
bert4keras使用beam search来解码文本标签生成时，如何优化性能呢

### ISSUE THREAD

**Issue by wuzhiye7** [bojone/bert4keras#293]
近期在用bert4keras生成模型做文本标签生成的实验，效果比做分类任务好些（标签自带了语义信息）。
遇到的问题是beam search的解码耗时较长（beam_size已经降低到2），分析了下每步的耗时，发现一次“ **scores = self.predict(inputs, output_ids, step, 'logits')  # 计算当前得分**  “ 就要60ms。有个想法是直接生成到标签的词集空间，会不会提高推理的性能？ 在解码时mask非标签的词或字，请教下，具体该如何修改源代码呢，麻烦指导下，谢谢

**Comment by bojone** [comment]
你是多标签生成还是单标签？

**Comment by wuzhiye7** [comment]
> 你是多标签生成还是单标签？

多标签

**Comment by bojone** [comment]
我感觉很难有什么明显提升速度的方法。

最快的方案应该是将每个标签当成一个词来生成，这样就最大程度上缩短了标签长度，从而堵加快了生成速度。

单纯mask掉非标签的token并不能带来什么提速效果。

**Comment by wuzhiye7** [comment]
> 我感觉很难有什么明显提升速度的方法。
> 
> 最快的方案应该是将每个标签当成一个词来生成，这样就最大程度上缩短了标签长度，从而堵加快了生成速度。
> 
> 单纯mask掉非标签的token并不能带来什么提速效果。

把标签当成词来生成，怕破坏了与文本之间的语义交互信息，不过值得藏试一下
